parastell/utils.py

```diff
@@ -1,70 +1,98 @@
 import yaml
-import math
 
 import numpy as np
+from scipy.ndimage import gaussian_filter
 
 m2cm = 100
 m3tocm3 = m2cm * m2cm * m2cm
 
 
-def normalize(vec_list):
-    """Normalizes a set of vectors.
+def downsample_loop(list, sample_mod):
+    """Downsamples a list representing a closed loop.
 
     Arguments:
-        vec_list (1 or 2D np array): single 1D vector or array of 1D vectors
-            to be normalized
+        list (iterable): closed loop.
+        sample_mod (int): sampling modifier.
+
     Returns:
-        vec_list (np array of same shape as input): single 1D normalized vector
-            or array of normalized 1D vectors
+        (iterable): downsampled closed loop
     """
-    if len(vec_list.shape) == 1:
-        return vec_list / np.linalg.norm(vec_list)
-    elif len(vec_list.shape) == 2:
-        return vec_list / np.linalg.norm(vec_list, axis=1)[:, np.newaxis]
-    else:
-        print('Input "vec_list" must be 1-D or 2-D NumPy array')
+    return np.append(list[:-1:sample_mod], [list[0]], axis=0)
 
 
-def expand_ang_list(ang_list, num_ang):
-    """Expands list of angles by linearly interpolating according to specified
-    number to include in stellarator build.
+def enforce_helical_symmetry(matrix):
+    """Ensures that a matrix is helically symmetric according to stellarator
+    geometry by overwriting certain matrix elements. Assumes regular spacing
+    between angles defining matrix.
 
     Arguments:
-        ang_list (list of double): user-supplied list of toroidal or poloidal
-            angles (rad).
-        num_ang (int): number of angles to include in stellarator build.
+        matrix (2-D iterable of float): matrix to be made helically symmetric.
 
     Returns:
-        ang_list_exp (list of double): interpolated list of angles (rad).
+        matrix (2-D iterable of float): helically symmetric matrix.
     """
-    ang_list = np.deg2rad(ang_list)
+    num_rows = matrix.shape[0]
+    num_columns = matrix.shape[1]
 
-    ang_list_exp = []
+    # Ensure rows represent closed loops
+    for idx in range(num_rows):
+        matrix[idx, -1] = matrix[idx, 0]
 
-    init_ang = ang_list[0]
-    final_ang = ang_list[-1]
-    ang_extent = final_ang - init_ang
+    # Ensure poloidal symmetry at beginning of period
+    matrix[0] = np.concatenate(
+        [
+            matrix[0, : int((num_columns - 1) / 2) + 1],
+            np.flip(matrix[0, : int(num_columns / 2)]),
+        ]
+    )
 
-    ang_diff_avg = ang_extent / (num_ang - 1)
+    # Ensure helical symmetry toroidally and poloidally by mirroring the period
+    # about both matrix axes
+    flattened_matrix = matrix.flatten()
 
-    for ang, next_ang in zip(ang_list[:-1], ang_list[1:]):
-        n_ang = math.ceil((next_ang - ang) / ang_diff_avg)
+    for idx, value in enumerate(
+        flattened_matrix[: int(len(flattened_matrix) / 2)], start=1
+    ):
+        flattened_matrix[-idx] = value
```

Comment:
Is this what it's doing?
```suggestion
    half_period = flattened_matrix[0:int(len(flattened_matrix) / 2)]
    flattened_matrix = np.concatenate([half_period, np.flip(half_period)])
```

Reply:
Something similar to the above snippet can be applied to the poloidal profile at the beginning of the period as well.